pkg/azuredx/adxauth/adxcredentials/builder.go

```diff
@@ -0,0 +1,171 @@
+package adxcredentials
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/grafana/azure-data-explorer-datasource/pkg/azuredx/adxauth/adxcredentials/internal/maputil"
+	"github.com/grafana/grafana-azure-sdk-go/azcredentials"
+	"github.com/grafana/grafana-azure-sdk-go/azsettings"
+)
+
+func FromDatasourceData(data map[string]interface{}, secureData map[string]string) (azcredentials.AzureCredentials, error) {
+	var credentials azcredentials.AzureCredentials
+	var err error
+
+	// Authentication types specific to the Azure Data Explorer datasource
+	if credentialsObj, err := maputil.GetMapOptional(data, "azureCredentials"); err != nil {
+		return nil, err
+	} else if credentialsObj != nil {
+		credentials, err = getFromCredentialsObject(data, credentialsObj, secureData)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
If OBO (1) is to stay permanently, it should be moved to SDK.
Legacy format (3) will have to stay here because it's ADX specific. Until we are comfortable to drop it.

Reply:
Let me consider adding OBO to SDK.

Reply:
I sent a PR to the SDK and introduced the new credential definitions (+ builder logic) there:
https://github.com/grafana/grafana-azure-sdk-go/pull/15

Reply:
Moved OBO to SDK, updated the PR.